src/wp-includes/l10n.php

```diff
@@ -1261,6 +1264,22 @@ function load_script_textdomain( $handle, $domain = 'default', $path = '' ) {
 		}
 	}
 
+	// Check if a custom path was registered for the domain.
+	$reg_path = $wp_textdomain_registry->get( $domain, $locale );
+	if ( $reg_path ) {
+		$reg_path = untrailingslashit( $reg_path );
+
+		$translations = load_script_translations( $reg_path . '/' . $handle_filename, $handle, $domain );
+		if ( $translations ) {
+			return $translations;
+		}
```

Comment:
Hmm this doesn't seem to make sense. Above and below we only check for `$md5_filename`. So this addition here adds some new behavior. Let's remove it.